src/VisualStudio/Core/Def/Implementation/EditAndContinue/VisualStudioActiveStatementProvider.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.EditAndContinue;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.VisualStudio.Debugger;
+using Microsoft.VisualStudio.Debugger.Clr;
+using Microsoft.VisualStudio.Debugger.Symbols;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.EditAndContinue
+{
+    [Export(typeof(IActiveStatementProvider)), Shared]
+    internal sealed partial class VisualStudioActiveStatementProvider : IActiveStatementProvider
+    {
+        public Task<ImmutableArray<ActiveStatementDebugInfo>> GetActiveStatementsAsync(CancellationToken cancellationToken)
```

Comment:
This can be called on any thread.

---
In reply to: [162466474](https://github.com/dotnet/roslyn/pull/24316#discussion_r162466474) [](ancestors = 162466474)

Reply:
Isn't all code considered free-threaded by default if it doesn't specify otherwise? Then comment shouldn't be needed.

---
In reply to: [162753416](https://github.com/dotnet/roslyn/pull/24316#discussion_r162753416) [](ancestors = 162753416,162466474)

Reply:
@tmat Code in the VS layer is presumed to be as free-threaded as VS is, which is to say "not".